bin/templates/cordova/lib/build.js

```diff
@@ -409,7 +442,29 @@ function parseOpts(options, resolvedTarget) {
             console.warn('Build option \'' + options[i] + '\' not recognized (ignoring).');
         }
     }
+    if (packageArgs.keystore && packageArgs.alias) {
+        var keystore = path.relative(ROOT, path.resolve(packageArgs.keystore));
+        ret.packageInfo = new PackageInfo(keystore, packageArgs.alias, packageArgs.keystorePassword,
+            packageArgs.password, packageArgs.keystoreType);
+    } else if (fs.existsSync(path.join(CORDOVAROOT, BUILD_CONFIG_FILE))) {
```

Comment:
I did not realize that this was the design all along - of not looking into parent directories. I like your idea of passing the path to the JSON file as a single argument. I'm less inclined to do others:
1. 'codova prepare' creating -singing.properties file: There will require a lot of platform specific logic being in cordova-lib and less than ideal.
2. Pass multiple command line parameters: Escaping complex arguments correctly on command line might be tricky and messy.

From what I can understand, this results in two changes:
1. cordova-lib to pass path to build.json. Perhaps find a way to version this change - so that older versions of the platforms do not see this unexpected arg.
2. Android build script consuming this build.json and reading the `android` section of it.